### PR TITLE
More robust error handling

### PR DESCRIPTION
**Pull request by hassy**

### CODE REVIEW

**Comment by hassy** [comment]
`TypeError: AwsHelper.pushToSocketServer is not a function` :/

**Comment by lennym** [comment]
@hassy It should be using `AwsHelper.pushResultToClient` - that's a bug.

**Comment by hassy** [comment]
No, not in this case (there are no packages to push, so we don't want to create an S3 bucket which `pushResultToClient` does)

**Comment by lennym** [comment]
The push to s3 is in a `forEach` on the items though, so if you send `items:[]` (which is the case) then nothing is stored.

Removing the method without a major version bump was an error though.

**Comment by tomgco** [comment]
@hassy Also it seems that AwsHelper.httpRequest has been removed https://github.com/numo-labs/aws-lambda-helper/compare/master@%7B1day%7D...master#diff-6d186b954a58d5bb740f73d84fe39073L307

**Comment by hassy** [comment]
Yup, on it 👍

File: lib/api_request.js [hunk 1]
```diff
@@ -131,9 +131,10 @@ module.exports = function api_request (params, callback) {
       AwsHelper.log.info({'message': 'No Hotel Ids Supplied in SNS Message'});
       var options = make_options(params);
       http_request(options, function (err, data) {
-        if (err) {
-          AwsHelper.log.info({err: err}, 'No Hotels Package results');
-          return callback(err);
+        if (err || !data.result || !Array.isArray(data.result) || data.result.length === 0) {
```
Comment (nelsonic) [inline lib/api_request.js]: that's quite a set of conditions. 👍

Comment (hassy) [inline lib/api_request.js]: Yup :) it's a mirror of https://github.com/numo-labs/lambda-ne-classic-package-provider/blob/567e65f16d58769e5c7d72aca8c1133fbe0ea518/lib/api_request.js#L97 - should maybe throw in an `isArray()` check in there as well.

Comment (nelsonic) [inline lib/api_request.js]: @hassy good plan. 👍

**Pull request merged**